src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -742,7 +756,7 @@ private void AccumulateMatchingCommands(IEnumerable<string> commandNames)
 
                             try
                             {
-                                CommandDiscovery.LookupCommandInfo(tempCommandName, this.MyInvocation.CommandOrigin, this.Context);
+                                CommandDiscovery.LookupCommandInfo(tempCommandName, CommandTypes.All, options, this.MyInvocation.CommandOrigin, this.Context);
```

Comment:
If we don't care about the case that `isPattern == true` (`-abbreviation` specified) and `moudleName` is not-null-or-empty, then we don't need to change this. In that case, the execution will fall in `TryAutoloadingModule`, which doesn't handle pattern at all 😄